demisto_sdk/commands/common/native_image.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Dict, List, Optional
+
+from pydantic import BaseModel
+
+from demisto_sdk.commands.common.constants import NATIVE_IMAGE_FILE_NAME
+from demisto_sdk.commands.common.handlers import JSON_Handler
+from demisto_sdk.commands.common.tools import extract_docker_image_from_text, get_dict_from_file
+
+json = JSON_Handler()
+logger = logging.getLogger('demisto-sdk')
+
+
+class NativeImage(BaseModel):
+    supported_docker_images: List[str]
+    docker_ref: str
+
+
+class IgnoredContentItem(BaseModel):
+    id: str
+    reason: str
+    ignored_native_images: List[str]
+
+
+class NativeImageConfig(BaseModel):
+    native_images: Dict[str, NativeImage]
+    ignored_content_items: List[IgnoredContentItem]
+
+
+def load_native_image_config(native_image_config_file_path: Optional[str] = None) -> Dict:
+    if not native_image_config_file_path:
+        native_image_config_file_path = f'Tests/{NATIVE_IMAGE_FILE_NAME}'
+
+    native_image_config_content, _ = get_dict_from_file(native_image_config_file_path)
+    return native_image_config_content
+
+
+def file_to_native_image_config(native_image_config_file_path: Optional[str] = None) -> NativeImageConfig:
+    return NativeImageConfig.parse_obj(load_native_image_config(native_image_config_file_path))
+
+
+def extract_native_image_version_for_server(native_image: str) -> str:
+    return native_image.replace('native:', '')
+
+
+def docker_images_to_native_images_support(native_images: Dict[str, NativeImage]) -> Dict[str, List[str]]:
+    """
+    Map all the docker images from the native image configuration file into the native-images which support it.
+
+    Examples:
+       {
+           "chromium": ["8.1", "8.2"],
+           "tesseract": ["8.1"]
+       }
+
+       chromium docker image is supported in both 8.1.0, 8.2.0 native images
+       while tesseract is only supported in 8.1.0
+    """
+    docker_images_to_native_images_mapping: Dict = {}
+
+    for native_image_name, native_image in native_images.items():
```

Comment:
are we recalculating this for every integration/script?
If so we should definitely move this logic to the config object, a singleton, and cache the result. This isnt changing between integrations

Reply:
made it a singleton